app/routes.py

```diff
@@ -1 +1,217 @@
+
+    query_params = {
+        "channel" : "#task-notification",
+        "text" : message
+    }
+
+    requests.post(PATH, params=query_params,headers={"Authorization": SLACK_API_KEY})
+
+@task_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def complete_incomplete_task(task_id):
+    task = validate_model(Task, task_id)
+
+    task.completed_at = datetime.utcnow()
+    
+    db.session.commit()
+
+    slack_bot_message(f"Someone just completed the task {task.title}")
+
+    return make_response({"task": task.to_dict()})
+
+@task_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_complete_task_incomplete(task_id):
+    task = validate_model(Task, task_id)
+
+    task.completed_at = None
+
+    db.session.commit()
+
+    return make_response({"task": task.to_dict()})
+        
+
+@task_bp.route("/<task_id>", methods=["DELETE"])
+def delete_task(task_id):
+    task = validate_model(Task, task_id)
+
+    db.session.delete(task)
+    db.session.commit()
+
+    return make_response({"details":f'Task {task_id} "{task.title}" successfully deleted'})
+
+################## GOAL ROUTES ######################
+
+
+@goal_bp.route("", methods=["POST"])
+def handle_goals():
+    request_body = request.get_json()
+    try:
+        new_goal = Goal(title=request_body["title"])
+    except:
+        abort(make_response({"details": "Invalid data"}, 400))
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return make_response({"goal":{
+            "id": new_goal.goal_id,
+            "title": new_goal.title
+        }}, 201)
+
+@goal_bp.route("", methods=["GET"])
+def read_all_goals():
+
+    goal = Goal.query.all()
+    goal_response = []
+    for goal in goal:
+        goal_response.append({
+            "id": goal.goal_id,
+            "title": goal.title
+        })
+
+    return jsonify(goal_response)
+
+@goal_bp.route("/<goal_id>", methods=["GET"])
+def read_one_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    return {"goal":{
+            "id": goal.goal_id,
+            "title": goal.title,
+        }}
+
+@goal_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    request_body = request.get_json()
+
+    goal.title = request_body["title"]
+
+    db.session.commit()
+
+    return make_response({"goal":{
+            "id": goal.goal_id,
+            "title": goal.title
+        }})
+
+@goal_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    db.session.delete(goal)
+    db.session.commit()
+
+    return make_response({"details":f'Goal {goal_id} "{goal.title}" successfully deleted'})
+
+@goal_bp.route("/<goal_id>/tasks", methods=["POST"])
+def add_task_ids_to_goal(goal_id):
+
+    goal = validate_model(Goal, goal_id)
+    request_body = request.get_json()
+    goal.tasks = [Task.query.get(task_id) for task_id in request_body["task_ids"]]
```

Comment:
Instead of calling `Task.query.get(task_id)`, we could call `validate_model(task_id)`. Since a POST request has to handle a request_body sent by the client, which could have errors in it, we need to validate the goal (like you already do) and validate the tasks.